From ebcd20553b63eb899b6983b5b205e374aba85e73 Mon Sep 17 00:00:00 2001
From: Lavanya Suresh <lavaks@codeaurora.org>
Date: Tue, 18 May 2021 12:56:15 +0530
Subject: [PATCH] hostapd: Fix HE chan switch command to use proper BW

VHT is not enabled when HE mode is indicated in channel
switch command, due to which VHT operation IE is missing
in beacon causing STA to connect in HT40 mode.

HE support for channel switch command patch is modified in
upstream commit causing this issue. It is reverted to the
previous version for fixing the same.

Fixes: I4bd28058adbc27 (hostap: upgrade hostapd to 59e9794c7d7e)

Signed-off-by: Lavanya Suresh <lavaks@codeaurora.org>
---
 src/ap/drv_callbacks.c | 38 ++++++++++++++++++++++----------------
 1 file changed, 22 insertions(+), 16 deletions(-)

--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -4714,6 +4714,11 @@ hostapd_switch_channel_fallback(struct h
 	iface->conf->ieee80211be = freq_params->eht_enabled;
 
 	/*
+	 * Resetting operating class to avoid referring previous values
+	 */
+	iface->conf->op_class = 0;
+
+	/*
 	 * cs_params must not be cleared earlier because the freq_params
 	 * argument may actually point to one of these.
 	 * These params will be cleared during interface disable below.
